### DISCUSSION TITLE
How to get metrics in v2?

### DISCUSSION THREAD

**Discussion by project0** [ory/hydra#3472] [Q&A]
Since there is no /metrics endpoint anymore and it was migrated to open telemetry, i have no clue how to gather the metrics. 
see also https://github.com/ory/hydra/issues/3468

I tried to set common OTEL env variables, but there are not metrics shipped. 
```
            - name: OTEL_METRICS_EXPORTER
              value: "otlp"
            - name: OTEL_EXPORTER_OTLP_ENDPOINT
              value: otel-collector.observability.svc.cluster.local:4317
```

Can someone help me to get metrics collection working? What do i need to configure?

**Comment by vinckr** [comment]
Hello @project0 

All Ory services expose an endpoint for snapshots of data collected by Prometheus. This endpoint is found at /metrics/prometheus for all projects. If you run the default configuration, it will be exposed at different ports depending on the service:

- Ory Kratos: http://{host}:4434/metrics/prometheus
- Ory Hydra: http://{host}:4445/metrics/prometheus
- Ory Keto: http://{host}:4468/metrics/prometheus
- Ory Oathkeeper http://{host}:9000/metrics/prometheus

You can find this information in the [Ory documentation](https://www.ory.sh/docs/self-hosted/operations/observability).
